extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQControllerTask.java

```diff
@@ -240,13 +241,14 @@ public TaskStatus runTask(final TaskToolbox toolbox) throws Exception
         injector.getInstance(Key.get(ServiceClientFactory.class, EscalatedGlobal.class));
     final OverlordClient overlordClient = injector.getInstance(OverlordClient.class)
                                                   .withRetryPolicy(StandardRetryPolicy.unlimited());
+    final TaskIdentitiesProvider taskIdentitiesProvider = injector.getInstance(TaskIdentitiesProvider.class);
```

Comment:
We should use the provider to get ALL instances of `TaskIdentitiesProvider` and then provide that as a list to the controller.

Reply:
There should be only one type of provider configured and in use, otherwise the usage will become too complicated.

Reply:
It could be implemented in a way that isn't so complicated. This is a tagging provider and it makes sense to have multiple providers depending on what the implemented finds useful. We do this with loggers and emitters and such. Does this need to be any different?